### PR TITLE
Add auto-updating of RHEL 7 x86_64

### PR DESCRIPTION
**Pull request by brycekahle**

### CODE REVIEW

File: .github/workflows/rhel7.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
+name: RHEL 7 x86_64
+on:
+  schedule:
+    - cron: "0 1 * * *"
+  workflow_dispatch: {}
+env:
+  BTFHUB_NO_SUDO: true
+jobs:
+  rhel7-update:
+    name: rhel 7 x86_64
```
Comment (val06) [inline .github/workflows/rhel7.yml:10]: let's make a convention for job name across all workflows

File: .github/workflows/rhel7.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+      - run: git config --system --add safe.directory $(pwd)
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+
+      - name: build elfutils
+        run: |
+          wget -nv https://sourceware.org/elfutils/ftp/0.190/elfutils-0.190.tar.bz2
```
Comment (val06) [inline .github/workflows/rhel7.yml:65]: is this a trusted source?

File: .github/workflows/rhel7.yml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+          mkdir build
+          cd build
+          cmake -D__LIB=lib -DDWARF_INCLUDE_DIR=/usr/include ..
+          make install
+          echo "/usr/local/lib" >> /etc/ld.so.conf.d/pahole.conf
+          ldconfig
+
+      - name: build btfhub
+        run: |
+          make
```
Comment (val06) [inline .github/workflows/rhel7.yml:92]: why is this different from other [workflows](https://github.com/DataDog/btfhub/blob/main/.github/workflows/amzn1.yml#L74-L77)?

**Review by val06 — CHANGES_REQUESTED** [review]
CR-ed

File: .github/workflows/rhel7.yml (hunk 1 above)
Comment (brycekahle) [reply .github/workflows/rhel7.yml:10]: This is my preferred convention, but I don't want to change all the other jobs in this PR.

File: .github/workflows/rhel7.yml (hunk 2 above)
Comment (brycekahle) [reply .github/workflows/rhel7.yml:65]: Yes. Did you look at the website? They are the authors.

File: .github/workflows/rhel7.yml (hunk 3 above)
Comment (brycekahle) [reply .github/workflows/rhel7.yml:92]: We don't need to explicitly specify `bash` unless we are using pipes.

**Review by val06 — APPROVED** [review]
consider creating reusable workflows or composite actions to reduce code duplication among workflows

**Pull request merged**